api/build.gradle

```diff
@@ -14,8 +14,8 @@ dependencies {
 
     annotationProcessor libraries.auto_value
 
-    signature "org.codehaus.mojo.signature:java17:1.0@signature"
-    signature "net.sf.androidscents.signature:android-api-level-14:4.0_r4@signature"
+    signature "org.codehaus.mojo.signature:java18:1.0@signature"
+    signature "net.sf.androidscents.signature:android-api-level-24:7.0_r2@signature"
```

Comment:
Okhttp uses API 21 signature and once before we had discussed following what they do to get Java 8 support in a way that follows the most used Android library. Is it possible to stick with that?

https://github.com/square/okhttp/blob/master/build.gradle#L153

Though if we just avoid `CompletableFuture` I think any signature there should be fine too.

Reply:
I don't have a strong opinion on this one. If `CompletableFuture` is the only thing we lose between 21 and 24 then I'm cool with that.